### ISSUE TITLE
received a non-200 status code from Readwise: code 400

### ISSUE THREAD

**Issue by NTU-Sherlock** [marcus-crane/october#134]
I have an issue with uploading in App purchased books with Macbook Kobo.desktop app.

---

<details><summary>System Details</summary><ul><li>Version: v1.10.0</li><li>Platform: darwin</li><li>Architecture: arm64</li><li>Onboarding Complete: true</li></details>
[20240927123114.json](https://github.com/user-attachments/files/17159064/20240927123114.json)

**Comment by NTU-Sherlock** [comment]
Actually, I am not so sure if the file I choose is correct, Kobo.sqlite.
I tried every other file, only this make sense.
however it returned errors.
File location is ~/Library/Application Support/Kobo/Kobo Desktop Edition/

**Comment by NTU-Sherlock** [comment]
Yes, I do have trouble with it. The syncing is not smooth, sometimes I have to disconnect and reconnect.
However, this book always failed to sync, that's why I have to tried October.

**Comment by marcus-crane** [comment]
I've installed Kobo Desktop Edition and I seem to have a similar issue (specifically that no highlights appear but I might find more as I get further in) so I'll do some debugging and see what's required to support the desktop version.

I'm not hugely surprised that it doesn't work out of the box as there are no guarantees that the Kobo's internal database file matches the desktop version.

The exact layout of the database mostly stays the same but does slightly change over time as new devices are released.

**Comment by NTU-Sherlock** [comment]
Hi Marcus,

Thanks for the response.

Do you think using iPhone Kobo app would make any difference?
Or I must connect with Kobo E-reader?

**Comment by NTU-Sherlock** [comment]
Hi Marcus,

I successfully Sync from Kobo official connections with Readwise.

This is what I did:
As shown in first post. The problem a specific books error- 一如既往, which is the one missing from readwise.

Then I review my notes on Kobo, I found out there is a typo "zhuyin"  (looks like ㄩ） in my note, so I deleted it.
I also delete some empty notes, (looks like " ") 

Then I tried re-sync again, the notes is import from Kobo to Readwise without a problem. 

Thank you very much!

**Comment by marcus-crane** [comment]
Hey there,

> Do you think using iPhone Kobo app would make any difference?
I've never tried using the iPhone app and then seeing what the resulting database looks like. October was never designed to work with anything but actual Kobo eReader devices. If other devices like the desktop app work, it's only be coincidence and not by design although I'm not opposed to adding support, depending on what is involved.

> I successfully Sync from Kobo official connections with Readwise.
Ah, great! I haven't looked into the issue any further as I have a lot of life stuff on at the moment so I haven't had much time at home

Thanks for the update and I'll post here when I get around to looking into the issue further.

**Comment by danicotillas** [comment]
I'm having exactly the same problem on Debian 12. Here my log:

`{"level":"info","msg":"Logger should be initialised now","time":"2024-11-13T15:39:46+01:00"}
{"level":"info","msg":"Backend is about to start up","time":"2024-11-13T15:39:46+01:00"}
{"level":"info","msg":"Successfully validated token against the Readwise API","time":"2024-11-13T15:40:03+01:00"}
{"batch_count":1,"highlight_count":246,"level":"info","msg":"Successfully parsed highlights","time":"2024-11-13T15:41:17+01:00"}
{"level":"error","msg":"Received a non-200 response from Readwise","response":"{\"note\":[\"This field may not be blank.\"]}","status_code":400,"time":"2024-11-13T15:41:18+01:00"}
{"batch_count":1,"highlight_count":246,"level":"info","msg":"Successfully parsed highlights","time":"2024-11-13T15:41:34+01:00"}
{"level":"error","msg":"Received a non-200 response from Readwise","response":"{\"note\":[\"This field may not be blank.\"]}","status_code":400,"time":"2024-11-13T15:41:35+01:00"}
{"device_id":"00000000-0000-0000-0000-000000000388","level":"info","msg":"Found an attached device","time":"2024-11-13T15:41:41+01:00"}
{"device_id":"00000000-0000-0000-0000-000000000388","level":"info","msg":"Found an attached device","time":"2024-11-13T15:41:41+01:00"}
{"batch_count":1,"highlight_count":246,"level":"info","msg":"Successfully parsed highlights","time":"2024-11-13T15:41:46+01:00"}
{"level":"error","msg":"Received a non-200 response from Readwise","response":"{\"note\":[\"This field may not be blank.\"]}","status_code":400,"time":"2024-11-13T15:41:47+01:00"}
{"batch_count":1,"highlight_count":246,"level":"info","msg":"Successfully parsed highlights","time":"2024-11-13T15:41:57+01:00"}
{"level":"error","msg":"Received a non-200 response from Readwise","response":"{\"note\":[\"This field may not be blank.\"]}","status_code":400,"time":"2024-11-13T15:41:57+01:00"}
{"level":"info","msg":"Shutting down. Goodbye!","time":"2024-11-13T15:41:59+01:00"}
`
